tunnel/tunnel.go

```diff
@@ -97,11 +102,16 @@ func NewServer(user, address, key string) (*Server, error) {
 		return nil, fmt.Errorf("error while reading key %s: %v", key, err)
 	}
 
+	if strings.HasPrefix(sshAgent, "$") {
+		sshAgent = os.Getenv(sshAgent[1:])
+	}
```

Comment:
What is the idea behind this? Is this common practice?

Reply:
From `man ssh_config`:

>**IdentityAgent**
>Specifies the UNIX-domain socket used to communicate with the authentication agent.
> This option overrides the SSH_AUTH_SOCK environment variable and can be used to select a specific agent.  Setting the socket name to none disables the use of an authentication agent.  If the string "SSH_AUTH_SOCK" is specified, the location of the socket will be read from the SSH_AUTH_SOCK environment variable.  Otherwise if the specified value begins with a `$' character, then it will be treated as an environment variable containing the location of the socket.

This snippet mirrors the behavior of OpenSSH except it does not have a special case for SSH_AUTH_SOCK (no $).

Reply:
Perfect, thanks!